docs: improve clarity in dependency injection guide (#68324)
angular/angular: adev/src/content/guide/di/creating-and-using-services.md

When you use `@Injectable({ providedIn: 'root' })` in your service, Angular:

- **Creates a single instance** (a singleton) for the entire application
- **Makes it available throughout your application** without additional configuration

> - **Enables tree-shaking** so the service is only included in your JavaScript bundle if it's actually used

Reviewer: ```suggestion
- **Enables tree-shaking** so Angular only includes the service in your JavaScript bundle if you actually use it
```
